examples/gradient_descent.py

```diff
@@ -0,0 +1,85 @@
+"""
+Gradient descent on a sphere.
+"""
+
+import numpy as np
+import matplotlib
+matplotlib.use("Agg")  # NOQA
+import matplotlib.pyplot as plt
+import matplotlib.animation as manimation
+
+from geomstats.hypersphere import Hypersphere
+import geomstats.visualization as visualization
+import geomstats.vectorization as vectorization
+
+SPHERE2 = Hypersphere(dimension=2)
+METRIC = SPHERE2.metric
+
+
+def gradient_descent(start,
+                     loss,
+                     grad,
+                     manifold,
+                     lr=0.05,
+                     max_iter=128,
+                     precision=1e-5):
+    x = start
+    for i in range(max_iter):
+        x_prev = x
+        euclidian = - lr * grad(x)
+        tangent_vec = manifold.projection_to_tangent_space(vector=euclidian,
+                                                           base_point=x)
+        x = manifold.metric.exp(base_point=x, tangent_vec=tangent_vec)[0]
```

Comment:
hm, the [0] does not look great. Is there a way to avoid this? Or should I change the vectorization strategy by enforcing the outputs of methods to be of the same ndim and shapes as the inputs (-> feature request)?

Reply:
Agreed - would be great if we didn't have to do this.